### PR TITLE
Add Basic GCP Support

### PR DESCRIPTION
**Pull request by JamesVorder**
Hey there! I love this little script, and it's made my life a lot easier. Thank you so much for sharing it!

Personally, I prefer to use GCP for my cloud projects, so I added some basic support for that here. The monitoring function is not in scope for my changes, only the basic backup functionality.

I'm not sure if you're even taking pull requests, but I thought this would make a nice addition! Please let me know if there is anything I can do to improve this PR so that it might be merged.

### CODE REVIEW

**Comment by rhasselbaum** [comment]
Thank you, James! Skimmed through quickly and looks good. I'll do a more thorough review this week.

**Comment by JamesVorder** [comment]
Heya, have you had a chance to give this a closer review?

File: config.yaml [hunk 1]
```diff
@@ -15,6 +18,7 @@ full_if_older_than: 1M
 remove_all_but_n_full: 3
 
 # Optional AWS config file with credentials.
+# Ignore if using GCP
```
Comment (rhasselbaum) [inline config.yaml]: Nit: period at the end.

File: README.md [hunk 2]
```diff
@@ -60,6 +62,10 @@ Alternatively, you can create the S3 bucket and IAM resources manually. Here are
 
 ### Set up the host
 
+If you are using GCP, please follow the [GCP Setup](./docs/GCP_SETUP.md) instructions before you continue.
```
Comment (rhasselbaum) [inline README.md]: The README would benefit from a bit of refactoring so someone trying to set up GCP doesn't have to flip back and forth between this and `GCP_SETUP.md`. Would you mind merging those instructions directly in here?

File: terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
+provider "google" {
+  project = "home-server-347301"
```
Comment (rhasselbaum) [inline terraform/main.tf]: I'm unfamiliar with Terraform's template language so I'll take your word for it that this works. 😊 But the project name here seems oddly specific.

**Review by rhasselbaum — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good, just a few small comments inline. Also, since Terraform supports AWS but it's not being used for it here, can you rename the `terraform` directory to `terraform-gcp` or similar? That way, impatient AWS users will be less likely to try to use it and get confused (I hope!).

File: terraform/main.tf (hunk 3 above)
Comment (JamesVorder) [reply terraform/main.tf]: Nice catch, this should have come from the tfvars too

**Review by rhasselbaum — APPROVED** [review]
Thanks. I think the README could still use some smoothing out, but I can take that up later. Good to merge!

**Pull request merged**